.github/workflows/ci.yml

```diff
@@ -0,0 +1,172 @@
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+name: CI
+
+jobs:
+  check:
+    name: Rust Check
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v2
+
+      - name: Install stable toolchain
+        id: toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          override: true
+
+      - name: Cache downloads and build
+        uses: actions/cache@v2
```

Comment:
lets just keep an eye on this. we had some issues with unbounded cache growth, and the caches are super slow. current full-build jobs take in the order of >40s, not sure how long the clean build times are in comparison.

Reply:
Caches are deleted when not used for a week.  Or if more than 5GiB of caches exist the oldest get deleted.

Cache times will always depend on the individual project and this will change as the project evolves.  In this case the caches help a lot with build time.  They're pretty conservative wrt to reuse how i've done them here.  This could possibly be done better to even re-use caches when deps change but then also clear out the unused items in the cache.  The current version is simple and won't grow unnecessarily large, which seems like a good starting point.